src/app/app.component.ts

```diff
@@ -261,14 +321,80 @@ export class AppComponent implements AfterViewInit {
     this.entrySelected = treeNodesSelected?.length > 0 ? treeNodesSelected[0] : undefined;
   }
 
+  async onToolbarOptionSelected(event) {
+    const customEvent = event as CustomEvent<ToolbarOption>;
```

Comment:
format document
If you type the event as CustomEvent<ToolbarOption> you shouldn't need to use another variable
async onToolbarOptionSelected(event: CustomEvent<ToolbarOption>) {...}